sink/sink

```diff
@@ -89,9 +97,38 @@ def expand(val, env):
     else:
         return val
 
+# Compare two JSON style values to see if one is contained in the other
+# For lists and dict make sure all items in expect are also present in result
+# For any other values a strict comparison is used
+def contains(expect, result, debug=True):
+    if isinstance(expect, list) and isinstance(result, list):
+        result = list(result)
+        for exp in expect:
+            while len(result):
+                res = result.pop(0)
```

Comment:
This will eat away all elements which don't match the first one in `expect` from the `result` copy. But they could still be matches for the second, third, etc. element in `expect`. Also, you now don't assert that each element in expect occurs only once in `list`, I thought this was important in case the same action was being done more than once in `result`?

For simple elements this would be correct even for repetitions of elements in `expect`:
```python
for i in expect:
    if  expect.count(i) != result.count(i):
        return False
else:
    return True
```
If `i` is an aggregate (list or dict) it still needs recursion of course, and then the case of "multiple instances of the same value" is not very well defined, as it's not clear whether an element which is a subset of the expected one should count as "same" or "different".

Can we somehow avoid the case that multiple identical entries in a list are significant? Perhaps by always appending the container name as a suffix to action labels, or something such?